File: cadwiki-nuget/cadwiki.NetUtils/Exceptions.vb
Option Strict On
Option Infer Off
Option Explicit On

Public Class Exceptions
    Public Shared Function GetStackTraceLines(ex As Exception) As List(Of String)
        Dim stackTrace As String = ex.StackTrace
        Dim seperator As Char() = CType(Environment.NewLine, Char())
        Return stackTrace.Split(seperator, StringSplitOptions.RemoveEmptyEntries).ToList()
    End Function

    Public Shared Function GetPrettyStringList(ex As Exception) As List(Of String)
        Dim list As New List(Of String)
        list.Add("-----------------------------------------------------------------------------")

        If (Not String.IsNullOrEmpty(ex.Message)) Then
            list.Add("Message : ".PadLeft(26) & ex.Message)
        End If
        If (Not String.IsNullOrEmpty(ex.StackTrace)) Then
            list.Add("StackTrace : ".PadLeft(26) & ex.StackTrace)
        End If
        If (Not String.IsNullOrEmpty(ex.InnerException.Message)) Then
            list.Add("InnerException Message : " & ex.InnerException.Message)
        End If
        'If (Not String.IsNullOrEmpty(ex.Message)) Then
        'End If

        list.Add("-----------------------------------------------------------------------------")
        Return list
    End Function
End Class

File: cadwiki-nuget/cadwiki.NetUtils/Exceptions.cs
using System;
using System.Collections.Generic;
using System.Linq;

namespace cadwiki.NetUtils
{

    public class Exceptions
    {
        public static List<string> GetStackTraceLines(Exception ex)
        {
            string stackTrace = ex.StackTrace;
            char[] seperator = Environment.NewLine.ToCharArray();
            return stackTrace.Split(seperator, StringSplitOptions.RemoveEmptyEntries).ToList();
        }

        public static List<string> GetPrettyStringList(Exception ex)
        {
            var list = new List<string>();
            list.Add("-----------------------------------------------------------------------------");

            if (ex != null && !string.IsNullOrEmpty(ex.Message))
            {
                list.Add("Message : ".PadLeft(26) + ex.Message);
            }
            if (ex.StackTrace != null && !string.IsNullOrEmpty(ex.StackTrace))
            {
                list.Add("StackTrace : ".PadLeft(26) + ex.StackTrace);
            }
            if (ex.InnerException != null && !string.IsNullOrEmpty(ex.InnerException.Message))
            {
                list.Add("InnerException Message : " + ex.InnerException.Message);
            }

            list.Add("-----------------------------------------------------------------------------");
            return list;
        }
    }
}